packages/inventory/src/ducks/inventory.ts

```diff
@@ -124,3 +126,26 @@ export const getInventoriesByServicePointsIdsFactory = createSelector(
     return values(inventoriesByIds);
   }
 );
+
+export const getInventoriesByExpiry = createSelector(
+  getInventoriesArray,
+  getActiveServicePoints,
+  (inventories, flag) => {
+    const inventoriesOfInterest: Inventory[] = [];
+    if (!flag) {
+      inventories.forEach((inventory) => {
+        if (new Date(inventory.deliveryDate) < new Date(inventory.accountabilityEndDate)) {
+          inventoriesOfInterest.push(inventory);
+        }
+      });
+      return inventoriesOfInterest;
+    } else {
+      inventories.forEach((inventory) => {
+        if (new Date(inventory.deliveryDate) > new Date(inventory.accountabilityEndDate)) {
+          inventoriesOfInterest.push(inventory);
+        }
+      });
+      return inventoriesOfInterest;
+    }
+  }
+);
```

Comment:
please add a doc string here.
ideally the selector should return the whole range of data if the filter is not provided.

Reply:
rename `flag` to `returnExpired`